lib/circuitry/provisioning.rb

```diff
@@ -0,0 +1,59 @@
+require 'circuitry/provisioning/queue_creator'
+require 'circuitry/provisioning/topic_creator'
+require 'circuitry/provisioning/subscription_creator'
+
+module Circuitry
+  module Provisioning
+    class << self
+      def provision_from_config(config, logger: Logger.new(STDOUT))
```

Comment:
I'd consider just calling this method `provision`.  There is no reason to add the `_from_config` suffix since you're not differentiating from other privisioning methods.